File: python/Fixed Heap Dynamic.py
# Fixed heap dynamic array
# This code demonstrates how to create a fixed-size dynamic array in Python
# The array is initialized with None, and then each element is set to its index multiplied by 2.

n = int(input("Enter size: "))
arr = [None] * n  # Allocated but fixed
for i in range(n):
    arr[i] = i * 2
print(arr)

File: c++/Fixed Heap Dynamic.cpp
//fixed heap dynamic memory allocation
// This code demonstrates the use of dynamic memory allocation in C++.
// It uses the new operator to allocate memory on the heap and delete to free it.
// It is a fixed-size array, meaning the size is determined at runtime but remains constant during the program's execution


#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Enter size: ";
    cin >> n;
    int* arr = new int[n]; // Memory allocated on heap
    for (int i = 0; i < n; i++) arr[i] = i * 2;
    for (int i = 0; i < n; i++) cout << arr[i] << " ";
    delete[] arr; // Free memory
}
